src/components/sideBar/menuItem/types.ts

```diff
@@ -1,14 +1,13 @@
-import { ReactNode } from "react";
 import { IconType } from "react-icons";
-import { Action, AnyAction, Dispatch } from "redux";
+import { Dispatch } from "redux";
 
 interface SectionItem {
-  path: string;
+  path?: string;
```

Comment:
making it optional doesn't mean you can Ignore it.

the mainSections.map(x => ...)

should use x.path if present.